docs/reference/cluster/delete-desired-balance.asciidoc

```diff
@@ -6,9 +6,13 @@
 
 NOTE: {cloud-only}
 
-Resets the desired balance and starts a new computation from the current allocation.
-This API may be used if desired balance computation diverged from the current state
-and is trying to move too many shards.
+Discards the current desired balance and computes a new desired balance starting from the current allocation of shards.
+This can sometimes help {es} find a desired balance which needs fewer shard movements to achieve, especially if the
+cluster has experienced changes so substantial that the current desired balance is no longer optimal without {es} having
+detected that the current desired balance will take more shard movements to achieve than needed. However, this API
+imposes a heavy load on the elected master node and may not always have the expected effect. Calling this API should
+never be necessary. Consider instead <<shards-rebalancing-heuristics,increasing the value of
+`cluster.routing.allocation.balance.threshold`>> to avoid excessive shard movements.
```

Comment:
I think we recommend calling this API after changing the threshold value? If that is the case, I am not sure how changing the threshold is an alternative. I agree that not manually invoking this API is the ideal state. But it seems we are still recommending this quite often in support cases. So is the phrase "should
never be necessary" a bit too strong?

Reply:
The strength is deliberate, we need to avoid a situation where folks just call this API at random (like they do with `POST _cluster/reroute?retry_failed`) without reporting it as a bug.

Reply:
IIUC, the doc change is intended to describe the ideal situation even we are not there yet. We use it as an opportunity to get feedback from users to improve it accordingly in future. That sounds ok to me.